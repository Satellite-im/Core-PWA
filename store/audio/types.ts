export interface AudioState {
  previousVolume: number
  inputVolume: number
  muted: boolean
  volume: number
  deafened: boolean
  sounds: {
    inboundMedia: number
    outboundMedia: number
    system: number
  }
}
