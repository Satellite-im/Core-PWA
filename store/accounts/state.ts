import { AccountsState } from './types'

const InitalAccountsState: AccountsState = {
  locked: true,
  error: '',
  pinHash: '',
  active: '',
  gasPrice: '',
  phrase: '',
  encryptedPhrase: '',
}

export default InitalAccountsState
