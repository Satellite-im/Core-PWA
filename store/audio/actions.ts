import type { AudioState } from './types'
import { Sounds } from '~/libraries/SoundManager/SoundManager'
import { $WebRTC } from '~/libraries/WebRTC/WebRTC'
import { ActionsArguments } from '~/types/store/store'

export default {
  /**
   * @method toggleMute
   * @description Toggles mute for outgoing audio
   * @example @click="toggleMute"
   */
  async toggleMute({
    state,
    commit,
    dispatch,
    rootState,
  }: ActionsArguments<AudioState>) {
    const { activeCall } = rootState.webrtc
    const call = activeCall && $WebRTC.getCall(activeCall.callId)

    dispatch('sounds/playSound', state.muted ? Sounds.MUTE : Sounds.UNMUTE, {
      root: true,
    })

    if (!call) {
      return
    }

    if (!state.muted) {
      await call.mute({ kind: 'audio' })
      commit('setMute', true)
      return
    }
    await call.unmute({ kind: 'audio' })
    commit('setMute', false)
  },
  /**
   * @method toggleDeafen
   * @description Toggles deafen for incoming audio
   * @example @click="toggleDeafen"
   */
  toggleDeafen({ commit, dispatch, state }: any) {
    const deafened = state.deafened
    if (!deafened) {
      dispatch('sounds/playSound', Sounds.DEAFEN, { root: true })
      dispatch('sounds/setMuteSounds', true, { root: true })
    } else {
      dispatch('sounds/setMuteSounds', false, { root: true })
      dispatch('sounds/playSound', Sounds.UNDEAFEN, { root: true })
    }
    commit('deafen')
  },
}
