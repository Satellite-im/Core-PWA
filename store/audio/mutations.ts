import { AudioState } from './types'

const mutations = {
  mute(state: AudioState) {
    state.muted = !state.muted
  },
  toggleMute(state: AudioState) {
    state.muted = !state.muted
  },
  deafen(state: AudioState) {
    const isDeafened = state.deafened
    state.previousVolume = isDeafened ? state.previousVolume : state.volume
    state.volume = isDeafened ? state.previousVolume : 0
    state.deafened = !isDeafened
  },
  setVolume(state: AudioState, volume: Number) {
    state.previousVolume = volume
    state.volume = volume
  },
  setSystemVolume(state: AudioState, systemVolume: Number) {
    state.sounds.system = systemVolume
  },
  setInputVolume(state: AudioState, inputVolume: Number) {
    state.inputVolume = inputVolume
  },
  setSoundLevel(
    state: AudioState,
    sound: {
      inboundMedia: Number
      outboundMedia: Number
      system: Number
    },
  ) {
    state.sounds = sound
  },
}

export default mutations
