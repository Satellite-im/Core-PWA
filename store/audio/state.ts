interface AudioState {
  muted: Boolean
  deafened: Boolean
  volume: Number
  inputVolume: Number
  previousVolume: Number
  sounds: {
    inboundMedia: Number
    outboundMedia: Number
    system: Number
  }
}

const InitalAudioState = (): AudioState => ({
  muted: false,
  deafened: false,
  volume: 100,
  inputVolume: 100,
  previousVolume: 0,
  sounds: {
    inboundMedia: 100,
    outboundMedia: 100,
    system: 100,
  },
})

export default InitalAudioState
