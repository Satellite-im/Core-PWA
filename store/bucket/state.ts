import { BucketState } from './types'

const InitialBucketState = (): BucketState => ({
  initialized: false,
  fileSystem: null,
})

export default InitialBucketState
