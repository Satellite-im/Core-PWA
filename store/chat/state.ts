import { ChatState } from './types'
import { ScrollDirections } from '~/types/chat/chat'

export const initialCurrentChat = {
  messages: [],
  page: 1,
  size: 10,
  hasNextPage: true,
  direction: ScrollDirections.TOP,
  isMessagesLoading: false,
  lastLoadedMessageId: '',
  isScrollOver: true,
  offset: 0,
  showOlderMessagesInfo: false,
}

const InitialChatState = (): ChatState => ({
  replies: [],
  chatTexts: [],
  files: {},
  countError: false,
  currentChat: initialCurrentChat,
  uploadProgress: {},
})

export default InitialChatState
