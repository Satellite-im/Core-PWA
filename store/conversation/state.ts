import { ConversationState } from './types'

export default function InitialConversationState(): ConversationState {
  return {
    id: '',
    type: 'friend',
    calling: false,
    participants: [],
  }
}
