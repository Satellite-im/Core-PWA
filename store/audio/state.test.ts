import { expect } from '@jest/globals'
import * as state from '~/store/audio/state'

describe('state.default', () => {
  test('0', () => {
    const result: any = state.default()
    expect(result).toMatchSnapshot()
  })
})
