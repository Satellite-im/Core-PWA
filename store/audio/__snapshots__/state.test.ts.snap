// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state.default 0 1`] = `
Object {
  "deafened": false,
  "inputVolume": 100,
  "muted": false,
  "previousVolume": 0,
  "sounds": Object {
    "inboundMedia": 100,
    "outboundMedia": 100,
    "system": 100,
  },
  "volume": 100,
}
`;
